### PR TITLE
Merging latest changes

### PR DESCRIPTION
**Pull request by kkatzen**
* Adding counties
* Restructuring Cards
* ReportWrapper to own file
* Consolidating control of FIPs
* Consolidating retrieval of Display names
* Aaron's data fetcher changes
* ...and more!

### CODE REVIEW

File: app/src/features/charts/DisparityBarChart.tsx [hunk 1]
```diff
@@ -102,7 +108,7 @@ function getSpec(
             x: { scale: "x", field: thickMeasure },
             y: { scale: "y", field: breakdownVar, band: 0.8 },
             text: {
-              signal: `format(datum["${thickMeasure}"], "") + "%"`,
+              signal: `isValid(datum["${thickMeasure}"]) ? datum["${thickMeasure}"] + "%" : "" `,
```
Comment (aaronsn) [inline app/src/features/charts/DisparityBarChart.tsx:111]: Can you add the isValid check to the simple bar chart too? I see NaN there for unknown

File: app/src/pages/ExploreDataPage.tsx [hunk 2]
```diff
@@ -218,48 +148,78 @@ function CarouselMadLib(props: {
...
-                    .map(([key, value]) => (
-                      // TODO - we may want to not have this alphabetized by ID by default
-                      <MenuItem key={value} value={key}>
-                        {value}
-                      </MenuItem>
-                    ))}
-                </Select>
-              </FormControl>
+              <>
+                {/* TODO - don't use this hack to figure out if its a FIPS or not*/}
```
Comment (aaronsn) [inline app/src/pages/ExploreDataPage.tsx:171]: yikes... this one may be worth fixing sooner rather than later

File: app/src/utils/Fips.ts [hunk 3]
```diff
@@ -58,61 +110,3158 @@ export const STATE_FIPS_MAP: Record<string, string> = {
...
+  "09005": "Litchfield",
+  "09007": "Middlesex",
+  "09009": "New Haven",
+  "09011": "New London",
+  "09013": "Tolland",
+  "09015": "Windham",
+  "10001": "Kent",
+  "10003": "New Castle",
+  "10005": "Sussex",
+  "11001": "District of Columbia",
```
Comment (aaronsn) [inline app/src/utils/Fips.ts:441]: Can we move this to the state level? It technically can go in either but I think it's more obvious at the state level

File: app/src/utils/Fips.ts [hunk 4]
```diff
@@ -58,61 +110,3158 @@ export const STATE_FIPS_MAP: Record<string, string> = {
...
-    return !this.isCounty() && !this.isUsa();
-  }
-
-  isCounty() {
-    return this.code.length === 5;
-  }
+      "60": "American Samoa",
+      "66": "Guam",
+      "69": "Northern Mariana Islands",
+      "72": "Puerto Rico",
```
Comment (aaronsn) [inline app/src/utils/Fips.ts:116]: Can you add Puerto Rico back in? We have ACS data for it

File: app/src/utils/Fips.ts [hunk 5]
```diff
@@ -1,9 +1,61 @@
...
+export const USA_DISPLAY_NAME = "the United States";
 // Fake FIPS code used to represent totals in USA for convenience
 export const USA_FIPS = "00";
 
+class Fips {
+  code: string;
+
+  // TODO- revisit optional name, this is brittle
+  constructor(code: string) {
+    if (!RegExp("^[0-9]{2}|[0-9]{5}").test(code)) {
```
Comment (aaronsn) [inline app/src/utils/Fips.ts:12]: Add "$" to match end of string

File: app/src/utils/Fips.ts [hunk 6]
```diff
@@ -1,9 +1,61 @@
 // TODO maybe rename this file since this string is out of place
 export const ALL_RACES_DISPLAY_NAME = "All races";
-export const USA_DISPLAY_NAME = "United States";
+export const USA_DISPLAY_NAME = "the United States";
```
Comment (aaronsn) [inline app/src/utils/Fips.ts:3]: Why did you add "the"? Most dashboards I've seen just use "United States", I think it's a little cleaner.

At the very least capitalize "the"

**Review by aaronsn — APPROVED** [review]
Haha this is a heck of a PR. Nice work :)

File: app/src/utils/Fips.ts (hunk 6 above)
Comment (kkatzen) [reply app/src/utils/Fips.ts:3]: katrina specifically requested this change... we can always change this later again as a group

Comment (aaronsn) [reply app/src/utils/Fips.ts:3]: Sure I don't care too much, but can you capitalize "The"?

File: app/src/pages/ExploreDataPage.tsx (hunk 2 above)
Comment (kkatzen) [reply app/src/pages/ExploreDataPage.tsx:171]: Yes definitely... I had something out to address it but I think doing it properly may be a larger refactor so I was gonna wait till everything else got checked in...

File: app/src/features/cards/DisparityBarChartCard.tsx [hunk 7]
```diff
@@ -1,63 +1,86 @@
...
+                {chartToggle === "percents" && (
+                  <DisparityBarChart
+                    data={props.dataset}
+                    thickMeasure={"population_pct" as VariableId}
+                    thinMeasure={(props.metricId + "_pct_of_geo") as VariableId}
+                    breakdownVar={props.breakdownVar as BreakdownVar}
+                  />
+                )}
+                {chartToggle !== "percents" && (
+                  // TODO- calculate actual ratio
```
[the remark this answers is not in the document]
Comment (kkatzen) [reply app/src/features/cards/DisparityBarChartCard.tsx:70]: I don't know... we should maybe discuss as a group? i added to frontend sync

File: app/src/pages/ExploreDataPage.tsx (hunk 2 above)
Comment (aaronsn) [reply app/src/pages/ExploreDataPage.tsx:171]: Sounds good, this is big enough as is :P

File: app/src/features/cards/CardWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  DATA_CATALOG_PAGE_LINK,
+} from "../../utils/urlutils";
+import { CardContent } from "@material-ui/core";
+import Typography from "@material-ui/core/Typography";
+import Divider from "@material-ui/core/Divider";
+
+function CardWrapper(props: {
+  datasetIds: string[];
+  titleText: string;
+  children: () => JSX.Element;
```
[the remark this answers is not in the document]
Comment (kkatzen) [reply app/src/features/cards/CardWrapper.tsx:16]: ah good catch
